src/components/checkbox/checkbox.test.jsx

```diff
@@ -0,0 +1,233 @@
+import Checkbox from './checkbox.jsx';
+
+describe('Checkbox', () => {
+  const name = 'You good?';
+  const requiredProps = {
+    id: 'fart',
+    label: name,
+  };
+
+  it('has defaults', () => {
+    const ref = createRef();
+    render(<Checkbox {...requiredProps} ref={ref} />);
+    expect(document.body).toMatchSnapshot();
+    expect(ref.current).toMatchSnapshot();
+  });
+
+  describe('className API', () => {
+    it('sets <input> classname', () => {
+      const className = 'kerfuffle';
+      render(<Checkbox {...requiredProps} className={className} />);
+      expect(screen.getByRole('checkbox', { name })).toHaveClass(className);
+    });
+  });
+
+  describe('cssContainer API', () => {
+    it('sets input container className', () => {
+      render(<Checkbox {...requiredProps} cssContainer="test-class" />);
+      expect(screen.getByRole('presentation')).toHaveClass('test-class');
+    });
+  });
+
+  describe('id API', () => {
+    it('sets the id', () => {
+      render(<Checkbox {...requiredProps} id="hello-mario" />);
+      expect(screen.getByRole('checkbox', { name })).toHaveAttribute('id', 'hello-mario');
+    });
+
+    it('is set on the ref', () => {
+      const ref = createRef();
+      render(<Checkbox {...requiredProps} ref={ref} id="hello-mario" />);
+      expect(ref.current.id).toBe('hello-mario');
+    });
+  });
+
+  describe('label API', () => {
+    it('can be set', () => {
+      render(<Checkbox {...requiredProps} label="Unique label" />);
+      expect(screen.getByRole('checkbox', { name: 'Unique label' })).toBeInTheDocument();
+    });
+  });
+
+  describe('name API', () => {
+    it('sets the name attribute', () => {
+      render(<Checkbox {...requiredProps} name="test-name" />);
+      expect(screen.getByRole('checkbox', { name })).toHaveAttribute('name', 'test-name');
+    });
+
+    it('is set on the ref', () => {
+      const ref = createRef();
+      render(<Checkbox {...requiredProps} name="unique-name" ref={ref} />);
+      expect(ref.current.name).toBe('unique-name');
+    });
+  });
+
+  it('sets the onblur handler', () => {
+    const onBlur = jest.fn();
+    render(<Checkbox {...requiredProps} onBlur={onBlur} />);
+    fireEvent.blur(screen.getByRole('checkbox', { name }));
+    expect(onBlur.mock.calls.length).toEqual(1);
+  });
+
+  describe('onChange API', () => {
+    it('sets the onChange handler', () => {
+      const onChange = jest.fn();
+      render(<Checkbox {...requiredProps} onChange={onChange} />);
+      userEvent.click(screen.getByRole('checkbox', { name }));
+      expect(onChange.mock.calls.length).toEqual(1);
+    });
+  });
+
+  describe('onFocus API', () => {
+    it('sets the onFocus handler', () => {
+      const onFocus = jest.fn();
+      render(<Checkbox {...requiredProps} onFocus={onFocus} />);
+      fireEvent.focus(screen.getByRole('checkbox', { name }));
+      expect(onFocus.mock.calls.length).toEqual(1);
+    });
+  });
+
+  describe('readonly api', () => {
```

Comment:
Since readOnly is boolean, please be exhaustive in the test coverage.

Reply:
My reasoning was that since the snapshot doesn't have the boolean value, adding the opposite case felt redundant.

I'm not 100% I grok the use case for snapshot testing when we also double-check the same values.

Reply:
456d2fb27de3ad8edc3fe2d8c286ecbbf483eaca

I added it (I'd rather have too much testing than not enough), I'm just not sure I understand the value of snapshot testing

Reply:
Snapshot tests are for defaults.

API tests are for particular component states. 

For example, right now the default is it is _not_ readonly. The markups shows that. That bit comes from `Checkbox.defaultProps.readOnly`. If it were changed to `true` then we update the snapshot. But without the API test coverage, we would _now_ be missing the coverage. And a developer most likely would assume there already exists coverage and all they need to do is double check the new snapshot and voila!